src/serialization/tokenizer.hpp

```diff
@@ -20,9 +20,19 @@
 
 #include "buffered_istream.hpp"
 
+#include <limits>
 #include <istream>
 #include <string>
 
+// use of illegal utf8 character for this purpose was added in a76be7ef1e921dabacd99f16ef440bf9673b8d98
+// has something to do with the result of the preprocessor whose format is essentially undocumented and I don't intend to delve into that as of writing this comment
+#define INLINED_PREPROCESS_DIRECTIVE_CHAR 254
```

Comment:
Does this really need to be a macro? It could just be a static constant.

Reply:
Agreed. constexprrrrr